### PR TITLE
[ET-1344] Fix 'Maximum function nesting' error

### PR DESCRIPTION
**Pull request by codingmusician**
### 🎫 Ticket

[ET-1344] 

### 🗒️ Description

Fixes this error:
`( ! ) Error: Maximum function nesting level of '256' reached, aborting! `

It turned out to be an endless loop when currency settings aren't ever set.

### 🎥 Artifacts 

https://i.imgur.com/GlqZjnh.png

### ✔️ Checklist
- [x] I've included a changelog entry in the readme.txt file. 
- [x] My code is tested. 
- [x] My code has [proper inline documentation](https://the-events-calendar.github.io/products-engineering/docs/code-standards/).

[ET-1344]: https://theeventscalendar.atlassian.net/browse/ET-1344?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by codingmusician** [comment]
Due to my local environment tests not working, I asked @redscar to help me update the snapshots.

File: src/Tickets/Commerce/Ticket.php [hunk 1]
```diff
@@ -270,7 +270,7 @@ public function get_ticket( $ticket_id ) {
 		}
 
 		/** @todo TribeCommerceLegacy: Remove the usage of Currency from Tribe Commerce totally, leave that behind. */
-		add_filter( 'tribe_get_option_ticket-commerce-currency-code', [ tribe( Legacy_Compat::class ), 'maybe_load_currency_code_from_tribe_commerce' ] );
+		add_filter( 'tribe_get_option_tickets-commerce-currency-code', [ tribe( Legacy_Compat::class ), 'maybe_load_currency_code_from_tribe_commerce' ] );
```
Comment (moraleida) [inline src/Tickets/Commerce/Ticket.php:273]: @codingmusician the filter name was correct. We need to intercept calls to `tribe_get_option_ticket-commerce-currency-code` (Tribe Commerce calls)

File: src/Tickets/Commerce/Legacy_Compat.php [hunk 2]
```diff
@@ -72,10 +72,8 @@ public function set_legacy_module_name( $name ) {
 	 */
 	public function maybe_load_currency_code_from_tribe_commerce( $value ) {
 
-		$currency = Currency::get_currency_code();
```
Comment (moraleida) [inline src/Tickets/Commerce/Legacy_Compat.php:75]: if we remove line 75 here we won't be able to get Tickets Commerce currency codes

**Review by moraleida — CHANGES_REQUESTED** [review]

File: src/Tickets/Commerce/Ticket.php (hunk 1 above)
Comment (moraleida) [reply src/Tickets/Commerce/Ticket.php:273]: If we don't, we get Tribe Commerce settings overriding Tickets Commerce settings for currency

**Review by moraleida — APPROVED** [review]

**Pull request merged**